src/test/java/StringManipulationTest.java

```diff
@@ -20,66 +29,87 @@ public void tearDown() {
     }
 
     @Test
-    public void testCount1() {
+    public void testCount1_NormalString() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
     @Test
-    public void testCount2() {
-        fail("Not yet implemented");
+    public void testCount2_EmptyString() {
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
     @Test
-    public void testCount3() {
-        fail("Not yet implemented");
+    public void testCount3_Null() {
+        manipulatedstring.setString(null);
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
     @Test
-    public void testCount4() {
-        fail("Not yet implemented");
+    public void testCount4_TabSeperatedWords() {
+        manipulatedstring.setString("This   is    my    string");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
     }
 
```

Comment:
How should removeNthCharacter(1, true/false) behave? Should the method return "" or null?